### ISSUE TITLE
How to set max tokens for Graphql Parser options?

### ISSUE THREAD

**Issue by rachana-sudhindradani** [kobylynskyi/graphql-java-codegen#1265]
I have been getting below error while trying to update to latest version. 
```
graphql.parser.exceptions.ParseCancelledException: More than 15,000 'grammar' tokens have been presented. To prevent Denial Of Service attacks, parsing has been cancelled.
    at graphql.parser.Parser.throwIfTokenProblems (Parser.java:467)
    at graphql.parser.Parser.lambda$getSafeTokenSource$4 (Parser.java:378)
    at graphql.parser.SafeTokenSource.callbackIfMaxExceeded (SafeTokenSource.java:61)
    at graphql.parser.SafeTokenSource.nextToken (SafeTokenSource.java:53)
```
I am unable to set the max token value to a number above 15000.
Can you please help me with what config to be added in Maven (I do not see it in the docs)? 

PS: Our Kotlin project uses Maven build tool. And we are trying to create our Graphql models in a library.

**Comment by tejas-sastry** [comment]
@kobylynskyi can we get some help here? @jxnu-liguobin any thoughts here?

**Comment by jxnu-liguobin** [comment]
I don't understand how it works. As mentioned in the previous discussion, I did not write a large schema, but instead split it into business modules and merge them.

**Comment by tejas-sastry** [comment]
@jxnu-liguobin how large can a single graphql file be to bypass this check? I'm trying to understand what the "token" is in this check. is it a type/input or is it more of individual words in the file?

**Comment by jxnu-liguobin** [comment]
> @jxnu-liguobin how large can a single graphql file be to bypass this check? I'm trying to understand what the "token" is in this check. is it a type/input or is it more of individual words in the file?

This involves Lexical analysis. I don't know about [antlr4](https://github.com/antlr/antlr4), but I think token is a character, tokens is a string

**Comment by MichaelaKranz** [comment]
I get the same exception:
[ERROR]
graphql.parser.exceptions.ParseCancelledException: Es wurden mehr als 15.000 'grammar' Token präsentiert. Um Denial-of-Service-Angriffe zu verhindern, wurde das Parsing abgebrochen
    at graphql.parser.Parser.throwIfTokenProblems (Parser.java:467)
    at graphql.parser.Parser.lambda$getSafeTokenSource$4 (Parser.java:378)
    at graphql.parser.SafeTokenSource.callbackIfMaxExceeded (SafeTokenSource.java:61)
    at graphql.parser.SafeTokenSource.nextToken (SafeTokenSource.java:53)
    at graphql.org.antlr.v4.runtime.BufferedTokenStream.fetch (BufferedTokenStream.java:169)
    at graphql.org.antlr.v4.runtime.BufferedTokenStream.sync (BufferedTokenStream.java:152)
    at graphql.org.antlr.v4.runtime.BufferedTokenStream.consume (BufferedTokenStream.java:136)
    at graphql.org.antlr.v4.runtime.Parser.consume (Parser.java:571)
    at graphql.parser.antlr.GraphqlParser.baseName (GraphqlParser.java:4045)
    at graphql.parser.antlr.GraphqlParser.name (GraphqlParser.java:4270)
    at graphql.parser.antlr.GraphqlParser.inputValueDefinition (GraphqlParser.java:1736)
    at graphql.parser.antlr.GraphqlParser.inputObjectValueDefinitions (GraphqlParser.java:2943)
    at graphql.parser.antlr.GraphqlParser.inputObjectTypeDefinition (GraphqlParser.java:2783)
    at graphql.parser.antlr.GraphqlParser.typeDefinition (GraphqlParser.java:778)
    at graphql.parser.antlr.GraphqlParser.typeSystemDefinition (GraphqlParser.java:353)
    at graphql.parser.antlr.GraphqlParser.definition (GraphqlParser.java:281)
    at graphql.parser.antlr.GraphqlParser.document (GraphqlParser.java:193)
    at graphql.parser.Parser.lambda$parseDocumentImpl$0 (Parser.java:224)
    at graphql.parser.Parser.parseImpl (Parser.java:294)
    at graphql.parser.Parser.parseDocumentImpl (Parser.java:228)
    at graphql.parser.Parser.parseDocument (Parser.java:156)
    at com.kobylynskyi.graphql.codegen.parser.GraphQLDocumentParser.readDocument (GraphQLDocumentParser.java:155)
    at com.kobylynskyi.graphql.codegen.parser.GraphQLDocumentParser.getDocumentFromSchemas (GraphQLDocumentParser.java:57)
    at com.kobylynskyi.graphql.codegen.GraphQLCodegen.generate (GraphQLCodegen.java:150)
    at io.github.kobylynskyi.graphql.codegen.GraphQLCodegenMojo.execute (GraphQLCodegenMojo.java:325)



The graphql.schema of our customer increases from 13.000 lines to 19.000 lines

https://stackoverflow.com/questions/70933025/more-than-15000-parse-tokens-have-been-presented

There is a by-default limit set of maxTokens as 15000, you can change it using below configuration:
ParserOptions.setDefaultParserOptions(ParserOptions.newParserOptions().maxTokens(<YOUR_NEW_VALUE>).build());


Can you involve a new option to configure the "maxTokens"?

**Comment by bowbahdoe** [comment]
I was going to try this out on a hasura generated graphql api and instantly ran into this. Its a sensible thing, but I want a way to turn it off

**Comment by bowbahdoe** [comment]
The more I think about it, it doesn't seem like this library's usage is susceptible to DDOS. Might be worth it to just remove the limit

**Comment by kobylynskyi** [comment]
Changed the default behavior by skipping token limit validation.
Users that want to keep it, can specify: `skipSchemaSizeLimit=false`
The change will be released in 5.9.0 soon.
